### DISCUSSION TITLE
Preferences migration(from android dataStore to Multiplatform Settings)

### DISCUSSION THREAD

**Discussion by PiotrPrus** [russhwolf/multiplatform-settings#122] [Q&A] [answered]
Is there a way to migrate user preferences (currently using dataStore on Android) to multiplatform-settings? I tried to use the same preferences name, but multiplatform requires `preferences_pb` suffix.

**Comment by PiotrPrus** [comment, marked as answer]
I figured it out. After reading this article:
https://medium.com/androiddevelopers/datastore-and-data-migration-fdca806eb1aa
When creating the datastore from the factory function, there is a migrations list parameter:
```
PreferenceDataStoreFactory.create(
            scope = CoroutineScope(ioDispatcher),
            migrations = listOf(MigrationUtils(appContext).myMigration)
        )
```

**Comment by russhwolf** [comment]
If you have a `val dataStore: DataStore<Preferences>`, you should be able to just do `val settings: FlowSettings = DataStoreSettings(dataStore)` without needing any sort of migration.

**Reply by PiotrPrus** [reply]
That's awesome! I think this could be in the documentation. Little section called "migration" would be really nice :)
